### PR TITLE
Added parents recursive endpoint

### PR DESCRIPTION
**Pull request by phorkish**
Added `/r/parents/<INSCRIPTION_ID>` and `/r/parents/<INSCRIPTION_ID>/<PAGE>` endpoints. This should take care of https://github.com/ordinals/ord/issues/3309.

### CODE REVIEW

**Comment by elocremarc** [comment]
@gmart7t2  would it be worth making this endpoint like your doing with the new children endpoint returning `Vec<InscriptionRecursive>` instead of `Vec<Ids>`? https://github.com/ordinals/ord/pull/3761

File: src/api.rs [hunk 1]
```diff
@@ -165,6 +165,13 @@ impl Output {
   }
 }
 
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct Parents {
+  pub ids: Vec<InscriptionId>,
+  pub more: bool,
+  pub page: usize,
+}
+
```
Comment (raphjaph) [inline src/api.rs:174]: On second thought let's just reuse the Inscriptions struct from above

File: src/subcommand/server.rs [hunk 2]
```diff
@@ -1868,6 +1873,28 @@ impl Server {
...
+    Path((inscription_id, page)): Path<(InscriptionId, usize)>,
+  ) -> ServerResult {
+    task::block_in_place(|| {
+      let child = index
+        .get_inscription_entry(inscription_id)?
+        .ok_or_not_found(|| format!("inscription {inscription_id}"))?;
+
+      let (ids, more) = index.get_parents_by_sequence_number_paginated(child.parents, page)?;
+
+      Ok(Json(api::Parents { ids, more, page }).into_response())
```
Comment (raphjaph) [inline src/subcommand/server.rs:1894]:
```suggestion
      Ok(Json(api::Inscriptions { ids, more, page }).into_response())
```

**Review by raphjaph — CHANGES_REQUESTED** [review]

**Comment by phorkish** [comment]
@raphjaph I've made the change you requested.

**Review by raphjaph — APPROVED** [review]
LGTM

**Pull request merged**